mloop_multishot.py

```diff
@@ -69,14 +121,20 @@ def cost_analysis(cost_key=(None,), maximize=True, x=None):
         cost_dict = cost_analysis(
             cost_key=config['cost_key'] if not config['mock'] else [],
             maximize=config['maximize'],
-            x=lyse.routine_storage.x if config['mock'] else None,
+            x=lyse.routine_storage.params[0] if config['mock'] else None,
         )
-        if not cost_dict['bad'] or not config['ignore_bad']:
+        if verify_globals(config) and (not cost_dict['bad'] or not config['ignore_bad']):

```

Comment:
It might be better to check these conditions in the opposite order. That would remove the need for inter-process communication whenever ignoring bad shots.

```suggestion
        if (not cost_dict['bad'] or not config['ignore_bad']) and verify_globals(config):
```